loki/expression/symbols.py

```diff
@@ -81,6 +81,10 @@ def source(self):
         """The :any:`Source` object for this expression node."""
         return self._source
 
+    @source.setter
+    def source(self, source):
+        self._source = source
```

Comment:
Eventually, this needs to simply be an attribute, not a property. But this will come with the mode to dataclasses.